src/components/menu/provider.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, Dispatch, FC, HTMLAttributes, SetStateAction, useState } from 'react'
+
+interface MenuContextValue {
+  triggerProps: HTMLAttributes<HTMLButtonElement>
+  popoverProps: HTMLAttributes<HTMLDivElement>
+  isOpen: boolean
+  setIsOpen: Dispatch<SetStateAction<boolean>>
+}
+
+const MenuContext = createContext<MenuContextValue | null>(null)
+
+const MenuProvider: FC = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const triggerProps = {
+    'aria-haspopup': true,
+    'aria-expanded': isOpen,
+    onClick: () => {
+      setIsOpen((prev) => !prev)
+    },
+  }
```

Comment:
**question:** Have you referred to https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/ and https://www.w3.org/WAI/ARIA/apg/patterns/menubar/ for guidance on what keyboard interactions we need to handle? If so, is that work currently in- or out-of-scope for this series of PRs?

Reply:
> question: Have you referred to https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/ and https://www.w3.org/WAI/ARIA/apg/patterns/menubar/ 

not yet, thanks I will look into that

> To be clear, I'm not suggesting keyboard interaction should be implemented in this PR. Just wondering if its been considered.

I thought it would be good to be included as well, I'll try to split it then